Program.cs

```diff
@@ -0,0 +1,111 @@
+﻿using System;
+
+namespace Homework3
+{
+    /// <summary>
+    /// Represents a logger for logging messages to a file.
+    /// </summary>
+    class Logger : IDisposable
+    {
+        private static Logger instance;
+        private string _logFilePath;
+        private StreamWriter _streamWriter;
+        private bool disposed = false;
```

Comment:
It's also common to prefix private fields with an underscore _. This is not strictly enforced by the language but is a widely adopted convention in the .NET community. However, you have used it above so you should follow conventions uniformly throughout the project.